packages/engine/src/datastore/arrow/ipc.rs

```diff
@@ -755,6 +757,8 @@ fn simulate_write_array_data(
 
 // ADD
 // Assumes buffer is at least the right length
+/// This should be safe because the assumption is that the [`simulate_record_batch_to_bytes`] 
+/// method was called to create the correct buffer length
```

Comment:
```suggestion
/// This should be safe because the assumption is that the [`simulate_record_batch_to_bytes`]
/// method was called to create the correct buffer length.
```